tests/unit/CoreHeadingTest.php

```diff
@@ -0,0 +1,486 @@
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => 'vivid-red-background-color',
+				'className'       => null,
+				'content'         => 'Heading with Background Color',
+				'cssClassName'    => 'wp-block-heading has-vivid-red-background-color has-background',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient'        => null,
+				'level'           => 2.0,
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => null,
+				'textAlign'       => null,
+				'textColor'       => null,
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_text_color() {
+		$block_content = '
+			<!-- wp:heading {"textColor":"vivid-red"} -->
+			<h2 class="wp-block-heading has-vivid-red-color has-text-color">Heading with Text Color</h2>
+			<!-- /wp:heading>
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'Heading with Text Color',
+				'cssClassName'    => 'wp-block-heading has-vivid-red-color has-text-color',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient'        => null,
+				'level'           => 2.0,
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => null,
+				'textAlign'       => null,
+				'textColor'       => 'vivid-red',
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_font_size() {
+		$block_content = '
+			<!-- wp:heading {"fontSize":"large"} -->
+			<h2 class="wp-block-heading has-large-font-size">Heading with Font Size</h2>
+			<!-- /wp:heading>
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'Heading with Font Size',
+				'cssClassName'    => 'wp-block-heading has-large-font-size',
+				'fontFamily'      => null,
+				'fontSize'        => 'large',
```

Comment:
.

Reply:
So `fontSize` is different than `typography` font size?

Reply:
Yes, In the context of the Gutenberg block editor:

- When you use a predefined font size (like "large"), it sets the fontSize attribute.
- When you input a custom font size (like "28px"), it typically goes into the `style.typography.fontSize` property.

Reply:
This is an _important_ distinction that changes most our tests/annotations, since it seems to hold true for _all_ predefined values vs style objects.

Here's `textColor` and `gradient` both working top-level:

```php
<!-- wp:heading {"style":{"elements":{"link":{"color":{"text":"var:preset|color|accent-4"}}}},"textColor":"accent-4","gradient":"gradient-3"} -->
<h2 class="wp-block-heading has-accent-4-color has-gradient-3-gradient-background has-text-color has-background has-link-color"></h2>
<!-- /wp:heading -->
```

![image](https://github.com/user-attachments/assets/30db4bab-26d1-47d2-b84b-6bcdd967d2ea)

Taking a cue from @Ta5r on today's call I think this entire class needs to be revised a bit:

1. Use the editor to create some block markup with the specific attributes we need to test. (e.g. we can probably combine `anchor`, `class-name`, and a few others).
2. Test specific `style` values separately from their top-level presets.
3. Review + update the annotations about working fields/ previously untested to account for the nuance between testing presets and explicit values.